File: ejercicios_csura/Semana01/soluciones/Ejercicio04.cs
using Semana01.soluciones;

namespace soluciones
{
    public class Ejercicio04 : IEjercicio
    {
        public void Ejecutar()
        {
            try
            {
                Console.WriteLine("Ingrese su nombre:");
                string nombre = (Console.ReadLine()).Trim();

                Console.WriteLine("Ingrese su edad:");
                int edad = int.Parse(Console.ReadLine().Trim());

                Console.WriteLine($"\nHola, {nombre}. Tienes {edad} años de edad.");
            }
            catch
            {
                Console.WriteLine("Error: Ingreso inválido. Asegúrese de ingresar un número entero para la edad.");
                return;
            }
        }
    }
}

File: ejercicios_csura/Semana03/soluciones/Ejercicio04.vb
Public Class Ejercicio04
    Implements IEjercicio

    Public Sub Ejecutar() Implements IEjercicio.Ejecutar
        Dim numero As Integer = 0
        Dim numeroValido As Boolean = False

        Do While Not numeroValido
            Try
                Console.WriteLine("Ingrese un número entero:")
                numero = CInt(Console.ReadLine().Trim())
            Catch
                Console.WriteLine($"Error: Ingrese un número entero válido.{vbCrLf}")
                Continue Do
            End Try

            numeroValido = True
        Loop

        If numero < 0 Then
            Console.WriteLine("El número ingresado es negativo.")
        ElseIf numero = 0 Then
            Console.WriteLine("El número ingresado es cero.")
        Else
            Console.WriteLine("El número ingresado es positivo.")
        End If
    End Sub
End Class
